lib/app/services/access_token_authenticator.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:async';
```

Comment:
Mostly just moved from gruene api core, I tried to improve logging a bit and removed global completer to simplify logic.